earthfile2llb/earthfile2llb.go

```diff
@@ -350,3 +350,42 @@ func GetTargetArgs(ctx context.Context, resolver *buildcontext.Resolver, gwClien
 	return args, nil
 
 }
+
+// ArgName returns the parsed name of an ARG command, the default value (if
+// any), and the state of the --required and --global flags.
+func ArgName(ctx context.Context, cmd spec.Command, isBase bool, explicitGlobal bool) (_ string, _ *string, isRequired bool, isGlobal bool, _ error) {
+	if cmd.Name != "ARG" {
+		return "", nil, false, false, errors.Errorf("ArgName was called with non-arg command type '%v'", cmd.Name)
+	}
+	opts, argName, dflt, err := parseArgArgs(ctx, cmd, isBase, explicitGlobal)
+	if err != nil {
+		return "", nil, false, false, errors.Wrapf(err, "could not parse opts for ARG [%v]", cmd)
+	}
+	return argName, dflt, opts.Required, opts.Global, nil
+}
+
+// ArtifactName returns the parsed name of a SAVE ARTIFACT command and its local
+// name (if any).
+func ArtifactName(ctx context.Context, cmd spec.Command) (string, *string, error) {
+	from, to, asLocal, ok := parseSaveArtifactArgs(cmd.Args)
+	if !ok {
+		return "", nil, errors.Errorf("could not parse opts for SAVE TARGET [%v]", cmd)
+	}
+	if to == "./" {
+		to = from
+	}
+	if asLocal == "" {
+		return to, nil, nil
+	}
+	return to, &asLocal, nil
+}
+
+// ImageNames returns the parsed names of a SAVE IMAGE command.
+func ImageNames(ctx context.Context, cmd spec.Command) ([]string, error) {
+	var opts saveImageOpts
+	args, err := parseArgs("SAVE IMAGE", &opts, getArgsCopy(cmd))
+	if err != nil {
+		return nil, errors.Wrapf(err, "invalid SAVE IMAGE arguments %v", cmd.Args)
+	}
+	return args, nil
+}
```

Comment:
Since these API functions are growing in scope, I would perhaps move them to a new file within the same package, to avoid mixing too many things in here. Possibly even a new sub-package?

Reply:
For the time being, I moved them to their own file. They rely pretty heavily on `parseArgs` and the like, and I don't feel like we're at the point where all of those functions should move to their own sub-package, yet.